src/lib/ledger_catchup/normal_catchup.ml

```diff
@@ -680,38 +680,52 @@ let run ~logger ~precomputed_values ~trust_system ~verifier ~network ~frontier
+                                Rose_tree.root subtree |> Cached.peek
                                 |> Envelope.Incoming.data
                               in
-                              let transition_hash =
-                                External_transition.Initial_validated
-                                .state_hash transition
+                              let children_transitions =
+                                List.concat_map
+                                  (Rose_tree.children subtree)
+                                  ~f:Rose_tree.flatten
+                              in
+                              let children_state_hashes =
+                                List.map children_transitions
+                                  ~f:(fun cached_transition ->
+                                    Cached.peek cached_transition
+                                    |> Envelope.Incoming.data
+                                    |> External_transition.Initial_validated
+                                       .state_hash )
                               in
                               [%log error]
                                 ~metadata:
-                                  [ ( "state_hash"
-                                    , State_hash.to_yojson transition_hash )
+                                  [ ( "state_hashes_of_children"
+                                    , `List
+                                        (List.map children_state_hashes
+                                           ~f:State_hash.to_yojson) )
+                                  ; ( "state_hash_of_root"
+                                    , State_hash.to_yojson
+                                      @@ External_transition.Initial_validated
+                                         .state_hash root_transition )
                                   ; ( "reason"
                                     , `String
                                         "no common ancestor with our \
                                          transition frontier" )
-                                  ; ( "protocol_state"
+                                  ; ( "protocol_state_of_root"
                                     , External_transition.Initial_validated
-                                      .protocol_state transition
+                                      .protocol_state root_transition
                                       |> Mina_state.Protocol_state
                                          .value_to_yojson ) ]
                                 "Validation error: external transition with \
                                  state hash $state_hash was rejected for \
-                                 reason $reason" ) ;
-                          Mina_metrics.(
-                            Counter.inc Rejected_blocks.no_common_ancestor
-                              (Float.of_int @@ List.length transitions)) ) ;
+                                 reason $reason" ;
```

Comment:
nit: Suggest mentioning in the message that its children were also rejected. `Validation error: external transition with state_hash $state_hash and its children were rejected for reason $reason`